### PR TITLE
Branch serialize map

### PR DESCRIPTION
**Pull request by yiwen101**
Add Class BidirectionalMap to keep track of the of link between person to lesson, and lesson to tasks.
Add serialize method of it.
Integrate the above into logic manager and model manager.
Have run smoke test to make sure the storage works

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -32,7 +32,8 @@ public class ModelManager implements Model {
     private State state = State.SCHEDULE; // Default state of app. Can be either SCHEDULE or STUDENTS
     private Person currentShowingPerson = null;
     private Lesson currentShowingLesson = null;
-    private Task currentEditingLesson = null;
```
Comment (pzl111) [inline src/main/java/seedu/address/model/ModelManager.java:35]: was this a typo from last time where currentEditingLesson is a Task type?

**Review by pzl111 — COMMENTED** [review]
LGTM, just have a small uncertainty

**Review by HollaG — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (yiwen101) [reply src/main/java/seedu/address/model/ModelManager.java:35]: Sorry, it is a typo, thank you for pointing that out.

Comment (yiwen101) [reply src/main/java/seedu/address/model/ModelManager.java:35]: I think it was a typo lastTime; I renamed it to currentShowingTask.

**Pull request merged**